sled-agent/src/services.rs

```diff
@@ -237,8 +241,22 @@ impl Error {
 
 impl From<Error> for omicron_common::api::external::Error {
     fn from(err: Error) -> Self {
-        omicron_common::api::external::Error::InternalError {
-            internal_message: err.to_string(),
+        match err {
+            err @ Error::RequestedConfigConflicts(_) => {
+                omicron_common::api::external::Error::invalid_request(&format!(
+                    "{:#}",
```

Comment:
These two don't have causes so I don't think it matters here.  I did put `{:#}` there by habit and future-proofing but I'll change them to `{}` so that one would not be misled.

Reply:
Fixed.